test/Index/Store/unit-one-sourcefile-remapped.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: rm -rf %t
+// RUN: %target-swift-frontend -index-store-path %t/idx -file-prefix-map %S=REMAPPED_SRC_ROOT -file-prefix-map $PWD=REMAPPED_BUILD_ROOT -file-prefix-map %t=REMAPPED_OUT_DIR -file-prefix-map $(cd ../../.. && pwd)=REMAPPED_BUILD_DIR %s -o %t/file1.o -typecheck
+// RUN: c-index-test core -print-unit %t/idx | %FileCheck -v %s
+
+// CHECK: file1.o-{{[A-Z0-9]*}}
```

Comment:
How fragile is the hash done here? I wonder if it should be validated to make sure it also doesn't take the absolute path into account?

Reply:
That's a good point, they should all be deterministic now, I need to update the tests with it

Reply:
Done, updated the tests - you can see the hash is stable between tests with the same remapping. Unfortunately, I don't believe this will work for PCMs since the PCMs themselves have their own hash (besides the unit hash) which can change based upon flags (see [here](https://cs.github.com/llvm/llvm-project/blob/2978d026819b82f77eef42d92b3b53919f375bd8/clang/lib/Frontend/CompilerInvocation.cpp#L4503)).